set.go

```diff
@@ -0,0 +1,293 @@
+package ksuid
+
+import "encoding/binary"
+
+// CompressedSet is an immutable data type which stores a set of KSUIDs.
+type CompressedSet []byte
+
+// Iter returns an iterator that produces all KSUIDs in the set.
+func (set CompressedSet) Iter() CompressedSetIter {
+	return CompressedSetIter{
+		content: []byte(set),
+	}
+}
+
+// Compress creates and returns a compressed set of KSUIDs from the list given
+// as arguments.
+func Compress(ids ...KSUID) CompressedSet {
+	c := byteLength + (len(ids) / 5)
+	b := make([]byte, 0, c)
+	return AppendCompressed(b, ids...)
+}
+
+// AppendCompressed uses the given byte slice as pre-allocated storage space to
+// build a KSUID set.
+//
+// Note that the set uses a compression technique to store the KSUIDs, so the
+// resuling length is not 20 x len(ids). The rule of thumb here is for the given
+// byte slice to reserve the amount of memory that the application would be OK
+// to waste.
+func AppendCompressed(set []byte, ids ...KSUID) CompressedSet {
+	if len(ids) != 0 {
+		Sort(ids) // TODO: this is the only dynamic memory allocation :'(
+		one := makeUint128(0, 1)
+
+		// The first KSUID is always written to the set, this is the starting
+		// point for all deltas.
+		set = append(set, ids[0][:]...)
+
+		timestamp := ids[0].Timestamp()
+		lastKSUID := ids[0]
+		lastValue := makeUint128FromPayload(ids[0][timestampLengthInBytes:])
+
+		for i := 1; i != len(ids); i++ {
```

Comment:
Curios - is it intentional this isn't the more traditional `i < len(ids)` or just a style thing?

Reply:
That's just a style thing.